blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ScheduleContext.java

```diff
@@ -141,10 +141,13 @@
     /** shard lock manager */
     private final ShardLockManager lockManager;
 
-    public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards) {
+    private final Clock clock;
+
+    public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards, Clock clock) {
```

Comment:
Maybe you should provide a convenience constructor that takes only 2 parameters, and the Clock is defaulted to DefaultClockImpl ? I think that would be the majority of the calls. Only in tests that you would provide a different Clock implementation.